Source/ActionController.swift

```diff
@@ -312,17 +325,33 @@ open class ActionController<ActionViewType: UICollectionViewCell, ActionDataType
     
     open func calculateContentInset() {
         // calculate content Inset
-        collectionView.layoutSubviews()
-        if let section = _sections.last, !settings.behavior.useDynamics {
-            let lastSectionIndex = _sections.count - 1
-            let layoutAtts = collectionViewLayout.layoutAttributesForItem(at: IndexPath(item: section.actions.count - 1, section: hasHeader() ? lastSectionIndex + 1 : lastSectionIndex))
-            contentHeight = layoutAtts!.frame.origin.y + layoutAtts!.frame.size.height
+        if collectionViewLayout.dynamicAnimator != nil {
+            
+            collectionViewLayout.shouldInvalidateLayout(forBoundsChange: CGRect(x: 0, y: 0, width: 0, height: 0))
+            
+            contentHeight = CGFloat(numberOfActions()) * settings.collectionView.cellHeightWhenDynamicsIsUsed + (CGFloat(_sections.count) * (collectionViewLayout.sectionInset.top + collectionViewLayout.sectionInset.bottom))
+            contentHeight += collectionView.contentInset.bottom
             
-            if settings.cancelView.showCancel && !settings.cancelView.hideCollectionViewBehindCancelView {
-                contentHeight += settings.cancelView.height
+            setUpContentInsetForHeight(view.frame.height)
+            
+            view.setNeedsLayout()
+            view.layoutIfNeeded()
+
+            
+        } else {
+            collectionView.layoutSubviews()
+            
+            if let section = _sections.last, !settings.behavior.useDynamics {
```

Comment:
This check `!settings.behavior.useDynamics` seems that is not needed anymore as you are checking it in the outside `if`